### ISSUE TITLE
Cannot create a category that contains (I guess) Unicode chars

### ISSUE THREAD

**Issue by mkruselj** [surge-synthesizer/surge#3673]
![image](https://user-images.githubusercontent.com/2393720/105041061-d57d8500-5a62-11eb-89ad-513a5af54dd0.png)

**Comment by mvf** [comment]
It's the same for patch names. `SurgeStorage::isValidName()` basically rejects everything that isn't ASCII letters or punctuation. The good news is that everything works when relaxing that check, including on Windows. So all that's needed is a proper Unicode check+whitespace trim.

**Comment by baconpaul** [comment]
should we replace isValidName with Surge::Storage::isValidUTF8 in that check? Easy to do and the kind of thing I'd be fine to push into 181 if you agree with that implementation.

**Comment by baconpaul** [comment]
https://github.com/surge-synthesizer/surge/blob/263e90920046b3e2bad5ba9be166cd6e4d481d37/src/common/SurgeStorage.cpp#L1811

that's the check i use before i push something into a vstgui text box (since that push with non utf8 core dumps vstgui linux right now).

**Comment by mvf** [comment]
Annoyingly, "valid UTF-8" is not quite enough. We have to actually look at the code points, e.g. for trimming whitespace and rejecting all-whitespace input. Better leave things ASCII for 1.8.1 IMHO. Apart from the slightly confusing error message it's not the end of the world.

**Comment by baconpaul** [comment]
OK! Thank you! And that’s certainly the easiest path yes :)

**Comment by mkruselj** [comment]
> Better leave things ASCII for 1.8.1 IMHO. Apart from the slightly confusing error message it's not the end of the world.

So now we're looking at 1.9.0 and this is in the milestone. Plenty of time until April. Could put this one in, then? 🙂

**Comment by baconpaul** [comment]
So in theory (in theory)

1. We could just skip the check at all
2. We could change the file open to be fs::path based (although it probably is at this point)
3. We could code defensively for if the file open write doesn't work

That means if you made your category name something like "..\..\..\Q:\Foo" we would just flag the file open error

It does mean that people could type crazy stuff into categories and we try a file open write, but this is user connected on their machine not network connected, so the bobby tables risk is low compared to a web app.

So basically what I'm saying is "fix this problem by removing the check and making sure we handle a failed file open properly then expect the user to type something reasonable" is a good approach.

**Comment by baconpaul** [comment]
Confirm that savePatchToPath bails out when it can't open a file, but does so silently. This also couples with the really awful ok cancel thing that I want to fix.

But for now you could

1. Modify savePatchToPath to prompt an error message with S:UI::pE before it does a return
2. Remove the isValidFileName check
3. Commit and close this issue

and it would all be fine
